### ISSUE TITLE
TEMPORAL-STATE-000: exact temporal-state semantics and lifecycle hardening program

### ISSUE THREAD

**Issue by Tristan-Stoltz-ERC** [Luminous-Dynamics/symthaea#3604]
## Purpose
Unify the temporal-backend lifecycle repair discovered during the NEUROARCH audit and make state semantics explicit before further temporal/cognitive claims are promoted.

Related root abstraction: #3586
Current correctness bugs: #3580 #3583 #3584 #3585 #3587 #3588
HierarchicalCfC prerequisites: #3574 #3577 #3578 #3579 #3581
Claim hygiene: #3582

## Why this is P0
The ordinary cognitive-loop temporal abstraction currently conflates several distinct operations:

```text
projected observation
exact evolution snapshot
reset
state mutation / perturbation
checkpoint / parameter state
```

This is not confined to an experimental backend.

Classic CfC is two-layer by default (`CfCNetworkConfig::default().num_layers = 2`), and both ordinary constructors retain that default. Yet `read_state()` returns only the last cell state while `inject(state)` writes that one vector into every cell. Therefore the current classic-CfC save/restore assumption is not exact on the default path.

HdcLtc has a strong engine-level `NetworkStateSnapshot` and internally pure prediction, but generic destructive-operation backup omits bridge-level `current_output`.

HierarchicalCfC has no exact evolution snapshot today and its projected `inject()` path is reset-style.

## Core theorem
For every backend/profile:

```text
ProjectedStateObservation
  != ExactEvolutionSnapshot
  != Reset
  != Perturbation
  != CheckpointState
  != TrainingContinuationState
```

No API, receipt, benchmark, or caller may infer one capability from another.

### P0A — containment before new capability
Stop known-invalid operations from silently mutating live state.

1. #3580: HCFC multi-horizon prediction must not use projected read/inject as restore.
2. #3583: HCFC consolidation must not run destructive replay while claiming live-state preservation without exact snapshot capability.
3. #3584: HCFC per-cycle training must not silently re-advance live online state while claiming sequence-correct/non-corrupting training.
4. #3585: recurrent masking must be capability-gated; unsupported backend requests return explicit `Unsupported`/`NotApplied`, never reset/replicate state.
5. #3587: classic CfC multi-horizon/consolidation code must stop calling last-layer projection/replication an exact restore.
6. #3588: HdcLtc destructive-operation backup must include wrapper-visible cache required by its claimed profile.

Containment is allowed to disable/skip an unsupported operation with explicit telemetry. It must not invent a lossy approximation and call it exact.

### P0B — explicit temporal-state API (#3586)
Introduce separate semantic surfaces for:
- projected observation;
- dynamic reset;
- exact evolution snapshot/restore;
- typed perturbation;
- prediction purity capability;
- historical-start training capability.

Migrate intentional reset callers away from zero-vector `inject()`.

Exact restore returns a typed `Result`; backend/profile mismatch cannot silently no-op.

### P1A — exact classic CfC inference snapshot (#3587 implementation)
Add an engine-owned versioned snapshot capturing every mutable value required to reproduce future inference under the qualified profile.

Minimum audit:
- ordered hidden state of every cell/layer;
- actual mutable tau arrays if they can change independently of checkpoint identity;
- semantic step/context counters if any affect output;
- online adaptation state when enabled, or explicitly reject that profile;
- subject/config/checkpoint/shape binding.

Do not include optimizer/training-only state in an inference-only snapshot unless it changes inference outputs.

### P1B — exact HdcLtc bridge snapshot (#3588)
Retain the existing engine `NetworkStateSnapshot` but wrap it with the bridge-level observable state required by the protected lifecycle profile, beginning with `current_output` and auditing counters/adaptive-dimension state by semantic necessity.

### P1C — exact HierarchicalCfC lifecycle (#3577 -> #3578 -> #3579)
1. deterministic realized configuration + Genesis construction;
2. exact inference snapshot/restore including actual tau state, hierarchy context and semantic counters;
3. truthful nominal-vs-realized tau observability and pure-prediction wrapper qualification.

Do not mix #3581 tau-algorithm selection into lifecycle correctness.

### P2 — re-enable qualified operations
Only after exact/capability surfaces exist:
- pure multi-horizon prediction;
- destructive replay with restored live inference state;
- historical-start detached training;
- backend-specific perturbations;
- effectome interventions;
- NEUROARCH LiquidIsland capabilities.

## Capability matrix — current state

| Backend | Projected observation | Engine exact snapshot | Wrapper exact snapshot | Pure prediction | Historical-start training | Generic recurrent mask |
|---|---|---|---|---|---|---|
| classic CfC | yes, last-cell projection | no qualified full snapshot | no | no | no | not qualified |
| HdcLtc | yes, cached projected output | yes | incomplete (#3588) | yes, internally | yes at engine snapshot profile | unsupported through generic projection |
| HierarchicalCfC | yes, slowest projected cell | no | no | no | no | unsupported |

This matrix is descriptive, not promotional. Capability entries become qualification claims only after executable tests bind the named profile.

## Conformance suite
Create one backend-neutral lifecycle suite with backend-specific expected capabilities.

Required laws:

### Observation purity
```text
observe(S); future(I) == future(I) from S
```

### Exact restore
```text
capture S
mutate/evolve
restore S
future(I,D) == reference future(I,D)
```

### Reset distinction
Reset must have an explicit reset profile and must not accept a projected observation merely to trigger reset semantics.

### Perturbation truthfulness
If a perturbation receipt says `Applied`, the requested perturbation actually occurred. Unsupported operations cannot degrade into reset.

### Prediction purity
For profiles advertising pure prediction:
```text
S -> probe -> future
==
S -> future
```
including all semantic evolution state and wrapper-observable state required by the profile.

### Training separation
For detached/historical-start training profiles:
```text
online inference evolution
!= training-start evolution
!= learned parameter state
```
Parameter updates may persist while live inference evolution is restored according to the explicit profile.

## Evidence receipts
Every lifecycle qualification receipt should bind:
- backend + capability profile/version;
- exact code/head;
- subject/config/checkpoint identity;
- snapshot schema/version;
- numerical execution profile;
- input/dt stream commitment;
- mutation/probe applied;
- pre/post state commitments;
- resource overhead;
- result: `Qualified | NotQualified | Inconclusive | InfrastructureIndeterminate`.

No broad backend label such as `CfC supports restore` without profile qualification.

## Resource accounting
Snapshot-based purity is not free. Record separately:
- capture bytes copied;
- restore bytes copied;
- snapshot retained bytes;
- perturbation bytes/ops;
- prediction forward work;
- instrumentation/trace overhead.

Do not hide exact-state copying cost inside model inference cost.

## Evidence impact policy
Historical outputs are not erased automatically.

Instead classify claim dependence:
- **unaffected**: static code/config fact not relying on state preservation;
- **raw retained / interpretation limited**: output exists but lifecycle assumptions were weaker than claimed;
- **rerun required**: claim materially requires state-pure prediction, preserved consolidation state, sequence-correct training, or typed perturbation;
- **invalid intervention label**: requested perturbation differed from executed reset/replication.

#3582 governs consciousness/Butlin interpretation separately.

## Stop rules
Until P0 containment lands:
- do not promote new classic-CfC/HCFC prediction-purity claims;
- do not use generic recurrent masking as a causal intervention on HdcLtc/HCFC;
- do not treat HCFC full-loop results as exact-state lifecycle evidence;
- do not infer effectome edges from interventions whose state restoration is unqualified.

Until P1 exact snapshots qualify:
- no LiquidIsland may advertise `ExactSnapshotRestore` for the affected backend/profile;
- no causal paired replay may claim exact same-start comparison;
- no architecture tournament may count unsupported lifecycle capabilities as candidate features.

## Acceptance
This umbrella closes only when:
1. ambiguous projected-state injection is removed/deprecated from evidence-sensitive callers;
2. intentional reset uses explicit reset APIs;
3. exact restore is typed and profile-bound;
4. every backend has truthful capability reporting;
5. P0 bugs have executable regression tests;
6. exact snapshot profiles are qualified for the backends used by NEUROARCH experiments;
7. affected historical claims are annotated/rerun according to dependency;
8. negative/null lifecycle results remain visible.

## Non-goals
- no architecture ranking;
- no consciousness conclusion;
- no hidden migration of production backend;
- no silent tau-algorithm change;
- no requirement that all backends support every capability.

**Comment by Tristan-Stoltz-ERC** [comment]
Implementation contract opened as draft PR #3605 (`TEMPORAL-STATE-001`). Exact head `9551d75e63525f7d42b374606714b936d19ee9ea`, base `4bad8af72ff775e7c869b6df83faba718a339a36`, exactly one commit / one documentation file.

#3605 is intentionally contract-only. It freezes the V1 lifecycle theorem, backend snapshot boundaries, typed perturbation semantics, conformance laws, evidence receipt shape, resource accounting, migration order, and NEUROARCH capability boundary. It qualifies no runtime capability by itself.

Implementation ordering remains P0 containment -> explicit API -> exact backend snapshots -> re-enable prediction/replay/training/perturbation under qualified profiles.

**Comment by Tristan-Stoltz-ERC** [comment]
P0 classic-CfC evidence tranche opened as draft PR #3613 (`6971592391c8e95dd089e4ac24ea68dcca4e4c22`). It is test-only and establishes the restore defect plus detached-clone containment before any production routing change.

This sharpens the P0/P1 split:
- P0 containment may use a detached clone for classic-CfC observational prediction, with clone cost measured separately;
- HCFC must remain explicit `Unsupported` for the same observational-prediction profile until #3577/#3578/#3579 provide deterministic construction and exact lifecycle state;
- HdcLtc keeps its internally pure prediction path and does not need clone containment;
- P1 classic-CfC exact inference snapshot should target only semantic inference state for the frozen-checkpoint profile, not training/optimizer/adaptation continuation state.

The P0 containment is therefore profile-specific rather than one generic workaround across backends.

**Comment by Tristan-Stoltz-ERC** [comment]
P0 implementation sequence is now concrete and deliberately staged:

1. #3613 — test-only regression proof: default classic CfC is multilayer, projected `read_state()/inject()` is lossy, detached clone preserves live state.
2. #3614 — additive `CfCNetwork::predict_forward_observational(&self, ...)` clone-based containment API. No cognitive-loop routing change yet.
3. after exact-head execution of #3613/#3614: wire classic-CfC multi-horizon prediction to the observational API; HCFC remains explicit Unsupported; HdcLtc keeps its native pure predictor.
4. implement `CfCInferenceSnapshotV1` as a separate P1 optimization/qualification tranche.
5. only after exact snapshot qualification consider replacing the clone containment path.

Important semantic distinction:
- observational prediction may clone the whole subject because no learning is meant to persist;
- consolidation/training cannot restore a whole-network clone because that would erase newly learned parameters. Consolidation needs an evolution-state-only backup whose restore preserves parameter updates.

Therefore `ObservationalPrediction` and `ExactEvolutionBackup` remain separate capabilities even on the same backend.

**Comment by Tristan-Stoltz-ERC** [comment]
Further lifecycle refinement: classic CfC needs **two different state capsules**, not one generic exact snapshot.

### A. `CfCInferenceSnapshotV1` — checkpoint-bound
Use for read-only prediction / paired replay where the checkpoint must remain unchanged.

Suggested identity guard can use existing deps only:
- domain-separated BLAKE3;
- exact serialized `CfCNetworkConfig` (bincode/serde);
- ordered `get_weights()` values encoded by exact IEEE-754 bits.

Restore rejects if current checkpoint/config fingerprint differs. This makes `capture -> mutate evolution only -> restore -> replay` exact for the named checkpoint profile.

### B. `CfCLocalEvolutionBackupV1` — parameter-mutation-tolerant
Use around consolidation/training where learned parameter changes are **supposed to persist** while the pre-training live hidden state is restored.

A checkpoint fingerprint MUST NOT be required here: training intentionally invalidates it. This capsule should restore only evolution state and be explicitly local/non-portable to its owner lineage, with strict layer/dimension validation and same-backend/profile checks.

A full-network clone restore is invalid for this use case because it would roll back the very learning consolidation intended to keep.

### C. adaptation/training continuation state — separate again
If online-adaptation continuation must be replayed exactly, include its adaptation statistics/optimizer state under a different named profile. Do not silently enlarge inference snapshots to cover it.

So the refined theorem is:

```text
CheckpointBoundInferenceSnapshot
!= LocalEvolutionBackupAcrossLearning
!= TrainingContinuationSnapshot
```

All three may contain overlapping hidden-state bytes, but they have different identity and persistence semantics.

**Comment by Tristan-Stoltz-ERC** [comment]
Added TEMPORAL-STATE-006 / #3621: prediction availability must not masquerade as prediction error.

This is now part of the P0 integration sequence because HCFC containment cannot safely be implemented by simply returning the existing zero-vector fallback: that fallback is intentionally interpreted as saturated prediction error and fans into downstream prediction/coherence/uncertainty/learning signals.

Refined sequence:
1. #3613 prove classic-CfC restore defect + clone containment;
2. #3614 qualify read-only classic-CfC observational prediction primitive;
3. #3621 introduce typed prediction availability/outcome and audit consumers;
4. route backends truthfully:
   - HdcLtc -> Available/native-pure profile;
   - classic CfC -> Available/clone-containment profile initially;
   - HCFC -> Unsupported until #3577/#3578/#3579;
5. only prediction outcomes marked Available contribute model prediction-error/coherence/uncertainty evidence;
6. #3617 exact checkpoint-bound snapshot may later replace clone containment without changing availability semantics;
7. #3618 separately repairs consolidation/live-state preservation across learning.

Epistemic rule:

```text
prediction unavailable
!= prediction maximally wrong
```

Reliability/capability failures remain separately observable.
